contracts/Compass.vy

```diff
@@ -164,19 +192,25 @@ def update_valset(consensus: Consensus, new_valset: Valset):
 # This makes calls to contracts that execute arbitrary logic
 # message_id is to prevent replay attack and every message_id can be used only once
 @external
-def submit_logic_call(consensus: Consensus, args: LogicCallArgs, message_id: uint256, deadline: uint256):
+@nonreentrant('lock')
+def submit_logic_call(consensus: Consensus, args: LogicCallArgs, paloma_address: bytes32, gas_estimation: uint256, message_id: uint256, deadline: uint256):
     assert block.timestamp <= deadline, "Timeout"
     assert not self.message_id_used[message_id], "Used Message_ID"
     self.message_id_used[message_id] = True
     # check if the supplied current validator set matches the saved checkpoint
     assert self.last_checkpoint == self.make_checkpoint(consensus.valset), "Incorrect Checkpoint"
-    # signing data is keccak256 hash of abi_encoded logic_call(args, message_id, compass_id, deadline)
-    args_hash: bytes32 = keccak256(_abi_encode(args, message_id, compass_id, deadline, method_id=method_id("logic_call((address,bytes),uint256,bytes32,uint256)")))
+    # signing data is keccak256 hash of abi_encoded logic_call(args, paloma_address, gas_estimation, message_id, compass_id, deadline)
+    args_hash: bytes32 = keccak256(_abi_encode(args, paloma_address, gas_estimation, message_id, compass_id, deadline, method_id=method_id("logic_call((address,bytes),bytes32,uint256,uint256,bytes32,uint256)")))
```

Comment:
Can we use `author_paloma_address` instead? I found compass very hard to understand as a newb, because variable names are sort of all over the place. At least with an added bit of information, it's a bit clearer WHAT paloma address should be passed here.

Reply:
OK. I will change the variable name.